pywhy_graphs/algorithms/tests/test_generic.py

```diff
@@ -228,3 +228,102 @@ def test_is_collider():
     S = {"A"}
 
     assert pywhy_graphs.inducing_path(admg, "Z", "Y", L, S)[0]
+
+
+def test_find_adc():
+    # K -> H -> Z -> X -> Y -> J <- K
```

Comment:
Can you add reasonings in the in-line comments for why these are/aren't valid ADC?

Reply:
Sure!